From c34ab2851f476a65aeb1b5fa28f4a6680b8b27bf Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 15 May 2025 12:12:19 +0300
Subject: [PATCH] [SW-227896] gaudi3: telem: increase no of retries

sometimes it is observed that updating telemetry
info in shared mmeory is failing and it succeds
with increase in retrials hence this patch increases
the retries to 25 and reduces delay to 2us.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: If01f583138becc2f1fdcc9d0e149efd786d7acb3
---

diff --git a/arch/gaudi3/common.c b/arch/gaudi3/common.c
index ff00685..25d2083 100644
--- a/arch/gaudi3/common.c
+++ b/arch/gaudi3/common.c
@@ -1382,8 +1382,8 @@
 	return false;
 }
 
-#define IPM_SHM_TIMEOUT		10
-#define IPM_SHM_RETRY		5
+#define TELEM_IPM_SHM_TIMEOUT		2
+#define TELEM_IPM_SHM_RETRY		25
 
 int telem_ipm_sh_mem_write_retry(struct pid_table *platdata, bool sleep)
 {
@@ -1391,7 +1391,7 @@
 	uint64_t *duration_ns = &platdata->power->throttling_duration;
 	uint32_t size = MEMBER_SIZE(struct hl_pid_shm, telem_power_throttling_duration);
 	uint32_t offset = offsetof(struct hl_pid_shm, telem_power_throttling_duration);
-	uint32_t retry = IPM_SHM_RETRY;
+	uint32_t retry = TELEM_IPM_SHM_RETRY;
 	int ret;
 
 	while (retry) {
@@ -1400,7 +1400,7 @@
 			return ret;
 
 		retry--;
-		(sleep) ? k_sleep(K_USEC(IPM_SHM_TIMEOUT)) : k_busy_wait(IPM_SHM_TIMEOUT);
+		(sleep) ? k_sleep(K_USEC(TELEM_IPM_SHM_TIMEOUT)) : k_busy_wait(TELEM_IPM_SHM_TIMEOUT);
 	}
 
 	return ret;
